src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataSource.cs

```diff
@@ -41,34 +47,43 @@ public int GetTotalCount()
                 return _totalCount.Value;
             }
 
-            var totalCount = 0;
+            _totalCount = 0;
 
+            var streamPosition = _stream.Position;
+            _stream.Position = 0;
+
+            var csvReader = new CsvReader(_streamReader, _configuration);
             try
             {
-                _csvReader.Read();
-                _csvReader.ReadHeader();
-                _csvReader.ValidateHeader<CsvPrice>();
+                csvReader.Read();
+                csvReader.ReadHeader();
+                csvReader.ValidateHeader<CsvPrice>();
             }
             catch (ValidationException)
             {
-                totalCount++;
+                _totalCount++;
             }
 
-            while (_csvReader.Read())
+            while (csvReader.Read())
```

Comment:
Shouldn't we dispose csv reader after using? Can't we use the same reader for get totals and read data?

Reply:
Unfortunately - no. CsvReader can only read forward and will not start from the start of a stream even if we will reset position. So both CsvReader and our data source can only be used once.

Reply:
Okay, but still we need to dispose csv helper after using

Reply:
Fixed. Also, I'm sorry, but I was wrong. CSV Reader can read from a specified position. But it can validate header only once. And while we use this validation as a workaround, it should throw an exception next time when we'll try to read data. Otherwise, everything will work (no exceptions or errors in callbacks), but data will not be read. It may confuse.